src/server.rs

```diff
@@ -60,3 +76,31 @@ pub fn handle_connection(
         }
     }
 }
+
+pub fn new_p2p_connection(

```

Comment:
I think this is in the wrong module- the server shouldn't have any knowledge of peers. We should just have a generic method that connects to an address and a port and call it from the p2p conn protocol